### PR TITLE
Support default queue type for vhost.

### LINKED ISSUE #258: Support setting a default queue type on vhosts
Greetings! 
 
Since RabbitMQ now supports a `default-queue-type` parameter for vhosts, I suggest supporting this parameter here. The resulting behavior is that queues created in a vhost that don't set a queue type are created with the specified default. Accepted values are `quorum`, `classic`, and `stream`. I'll take a shot at a pull request, if I have time to do so and no one beats me to it.

Useful context: https://www.rabbitmq.com/rabbitmqctl.8.html#Virtual_hosts

### PR DESCRIPTION
**Pull request by mqhenning**
Fixes #258, depends on https://github.com/rabbitmq/rabbitmq-server/pull/7739 to ship in `3.11.12`.

### CODE REVIEW

File: vhosts.go [hunk 1]
```diff
@@ -161,6 +163,8 @@ type VhostSettings struct {
 	Description string `json:"description"`
 	// Virtual host tags
 	Tags VhostTags `json:"tags"`
+	// Type of queue to create in virtual host when unspecified
+	DefaultQueueType string `json:"defaultqueuetype,omitempty"`
```
Comment (michaelklishin) [inline vhosts.go:167]: Why is the JSON field not `default_queue_type` here? It is in the management UI's code (the API does not explicitly validate the keys in this case).

**Review by michaelklishin — CHANGES_REQUESTED** [review]

File: vhosts.go (hunk 1 above)
Comment (vashibhavin) [reply vhosts.go:167]: The inconsistency can be demonstrated using the following test. 

1. Create vhost with `defaultqueuetype`

```
> curl 'https://localhost:5672/api/vhosts/testvhost' -X 'PUT' -u mquser --data-raw '{"name":"testvhost","description":"","tags":"","defaultqueuetype":"quorum"}'
Enter host password for user 'mquser':
```
```
> curl -s 'https://localhost:5672/api/vhosts/testvhost' -u mquser  | jq
Enter host password for user 'mquser':
{
  "name": "testvhost",
  "description": "",
  "tags": [],
  "default_queue_type": "quorum",
  "metadata": {
    "default_queue_type": "quorum",
    "description": "",
    "tags": []
  },
  "tracing": false,
  "cluster_state": {
 ------
  }
}
```

2. Create vhost with `default_queue_type`
```
>curl 'https://localhost:5672/api/vhosts/testvhost1' -X 'PUT' -u mquser --data-raw '{"name":"testvhost1","description":"","tags":"","default_queue_type":"quorum"}'
Enter host password for user 'mquser':
```

```
> curl 'https://localhost:5672/api/vhosts/testvhost1' -u mquser  | jq
Enter host password for user 'mquser':
{
  "name": "testvhost1",
  "description": "",
  "tags": [],
  "default_queue_type": "undefined",
  "metadata": {
    "description": "",
    "tags": []
  },
  "tracing": false,
  "cluster_state": {
------  
  }
}
```

Comment (mqhenning) [reply vhosts.go:167]: Created a [bug](https://github.com/rabbitmq/rabbitmq-server/issues/7734) for rabbitmq-server.

Comment (michaelklishin) [reply vhosts.go:167]: Thank you. I have fixed the issue in https://github.com/rabbitmq/rabbitmq-server/pull/7739, let's wait for `3.11.12` to ship.

**Comment by michaelklishin** [comment]
This now depends on https://github.com/rabbitmq/rabbitmq-server/pull/7739 to ship in `3.11.12`.

**Review by michaelklishin — APPROVED** [review]

**Comment by michaelklishin** [comment]
@mqhenning would you mind changing the JSON key name to match what it will be in `3.11.12`?

**Comment by mqhenning** [comment]
> @mqhenning would you mind changing the JSON key name to match what it will be in `3.11.12`?

I suppose this won't be usable on earlier versions, but I'm happy to do away with the odd inconsistency. Done!

**Comment by michaelklishin** [comment]
Will rebase and merge once the Docker image build of `3.11.13` is out. Should be some time early next week.

**Comment by michaelklishin** [comment]
RabbitMQ `3.11.13` [has made it to Docker Hub](https://hub.docker.com/_/rabbitmq).

**Pull request merged**